scripts/build-release-artifacts.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script automates the building of artifacts for a release.
+#
+# Reference:  docs/releasing.md
+
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+RELEASE_TAG="${1-}"
+if [[ ! "${RELEASE_TAG}" ]]; then
+  >&2 echo "usage: $0 <release tag of the form v[0-9]+.[0-9]+.[0-9]+>"
+  exit 1
+fi
+
+ROOT_DIR="$(cd "$(dirname "$0")/.." ; pwd)"
+RELEASE_VERSION="${RELEASE_TAG:1:${#RELEASE_TAG}-1}"
+
+pushd "${ROOT_DIR}"
+  # Build release artifacts for kubefedctl
+  make kubefedctl
+  pushd "${ROOT_DIR}/bin"
+    # The binary is currently built with a platform suffix, but should be archived without it.
+    tar cvzf kubefedctl.tgz --transform='flags=r;s|kubefedctl-linux|kubefedctl|' kubefedctl-linux
+    sha256sum kubefedctl.tgz > kubefedctl.tgz.sha
+    mv kubefedctl.tgz kubefedctl.tgz.sha "${ROOT_DIR}/"
+  popd
+
+  # Build release artifacts for the helm chart
+  pushd "${ROOT_DIR}/charts"
+    # Update the image tag for the chart
+    sed -i.backup "s+\(  tag: \)canary+\1${RELEASE_TAG}+" federation-v2/values.yaml
+
+    # Update the chart version
+    sed -i.backup "s+\(version: \).*+\1${RELEASE_VERSION}+" federation-v2/Chart.yaml
+
+    helm package federation-v2
+
+    # Update the repo index (will need to be committed)
+    helm repo index . --merge index.yaml --url="https://github.com/kubernetes-sigs/federation-v2/releases/download/${RELEASE_TAG}"
+
+    sha256sum "federation-v2-${RELEASE_VERSION}.tgz" > "federation-v2-${RELEASE_VERSION}.tgz.sha"
+    mv federation-v2-${RELEASE_VERSION}.tgz* "${ROOT_DIR}/"
+
+    # Revert the chart changes (should not be committed)
+    mv federation-v2/values.yaml.backup federation-v2/values.yaml
```

Comment:
Not part of this PR, but I'm thinking it might make sense to commit these changes as part of the tagged release, possibly once we go beta. The idea is that you can checkout a tagged release and deploy it without having to download any artifacts. This would probably be for dev use primarily.

Reply:
I don't know that what you're suggesting makes sense in the context of a 'tag'.  A branch, maybe, since that's a persistent context that can be updated, but a tag is just a persistent alias for a commit hash.

Reply:
Right, and that commit object would point to code that contains the right version of stuff. It could either be based off `master` as we do today, but that would require a commit dance if we wanted to keep using the `canary` tag. Or a new/temporary branch named the same (or similarly) as the tag.